src/test/components/MenuButtons.test.js

```diff
@@ -278,6 +285,31 @@ describe('app/MenuButtons', function() {
       expect(queryPreferenceCheckbox()).toBeFalsy();
     });
 
+    it('can publish and revert', async () => {
+      const { profile } = createSimpleProfile();
+      const {
+        getPublishButton,
+        getPanelForm,
+        clickAndRunTimers,
+        resolveUpload,
+        getState,
+      } = setup(profile);
+      clickAndRunTimers(getPublishButton());
+      fireEvent.submit(getPanelForm());
+      resolveUpload('SOME_HASH');
+
+      const revertButton = await screen.findByText(/revert/i);
+      expect(getDataSource(getState())).toBe('public');
+      expect(getHash(getState())).toBe('SOME_HASH');
+      expect(document.body).toMatchSnapshot();
```

Comment:
Question: `document.body` is surprising here. Could you explain why do this rather than the component container like we have in the past?

Reply:
For me, this is part of the simplification when using `screen`. React testing library by default creates a `<div>` inside `document.body` and renders there. This `div` is what the returned `container` is.
We never change that default.

The queries that we can use on `screen` are all [bound to `document.body`](https://testing-library.com/docs/dom-testing-library/api-queries#screen). So it makes sense that we match `document.body`. Also there isn't anything handy on `screen` to access the react container, so we'd still need to return and forward the result of `render` in all our setup functions.

Always using `document.body` instead of `container` or `container.firstChild` could also bring more consistency, at the price of more nested snapshots. Indeed we were mostly using `container.firstChild`, but `container` when dealing with components rendering a fragment, and even possibly `document.body` when dealing with portals (I haven't checked).

Using it here wasn't really thought through that much :) more I was trying it and see if this is a viable option. Now I think it is but we can discuss this more broadly of course.

Hope it makes sense!